speech/grpc/transcribe_streaming.py

```diff
@@ -90,19 +90,22 @@ def _audio_data_generator(buff):
         yield b''.join(data)
 
 
-def _fill_buffer(audio_stream, buff, chunk):
+def _fill_buffer(audio_stream, buff, chunk, stoprequest):
     """Continuously collect data from the audio stream, into the buffer."""
     try:
-        while True:
+        while not stoprequest.isSet():
             buff.put(audio_stream.read(chunk))
     except IOError:
         # This happens when the stream is closed. Signal that we're done.
         buff.put(None)
 
+    audio_stream.stop_stream()
+    audio_stream.close()
+
```

Comment:
No, this happens because the stream is stopped and the thread hangs when this thread is still in the loop.